src/GPS.cpp

```diff
@@ -0,0 +1,73 @@
+#include "GPS.hpp"
+#include <nmea0183/GPS.hpp>
+
+using namespace nmea0183;
+using namespace marnav;
+using namespace std;
+using namespace gps_base;
+
+GPS::GPS(Driver& driver)
+    : m_driver(driver)
+{
+}
+
+gps_base::GPS_SOLUTION_TYPES GPS::getPositionType(
+    marnav::utils::optional<marnav::nmea::mode_indicator> mode)
+{
+    // TODO: check returns
+    switch (*mode) {
+        case marnav::nmea::mode_indicator::invalid:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::autonomous:
+            return GPS_SOLUTION_TYPES::AUTONOMOUS_2D;
+        case marnav::nmea::mode_indicator::differential:
+            return GPS_SOLUTION_TYPES::DIFFERENTIAL;
+        case marnav::nmea::mode_indicator::estimated:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::manual_input:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::simulated:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::data_not_valid:
+            return GPS_SOLUTION_TYPES::NO_SOLUTION;
+        case marnav::nmea::mode_indicator::precise:
```

Comment:
`precise` is invalid ?

Reply:
I don't know what would suit better here, for **estimated**, for **manual_input** and for **simulated**.
Should I add these types to GPS_SOLUTION_TYPES?